### PR TITLE
Eas0

### PR DESCRIPTION
**Pull request by schweitzpgi**
Get lit tests working for fir

### CODE REVIEW

File: test-lit/fir/fir-ops.fir [hunk 1]
```diff
@@ -0,0 +1,341 @@
+// Test the FIR operations
```
Comment (sscalpone) [inline test-lit/fir/fir-ops.fir]: Maybe a comment for our future selves -- Is this a round-trip test?  Should all new fir operations be added to this test?  Is it complete right now?

**Comment by sscalpone** [comment]
Nice!

File: include/fir/Dialect/FIROps.td [hunk 2]
```diff
@@ -117,6 +120,7 @@ class fir_SimpleOp<string mnemonic, list<OpTrait> traits>
   }];
 }
 
+// Base builder for allocate operations
```
Comment (jeanPerier) [inline include/fir/Dialect/FIROps.td]: Would using `let description =[{Base builder for allocate operations}]` here instead of `//` comments make sens here ?

File: tools/bbc/bbc.cpp [hunk 3]
```diff
@@ -147,7 +148,14 @@ void convertFortranSourceToMLIR(
 
   if (EmitLLVM) {
     pm.addPass(fir::createFIRToLLVMPass(nameUniquer));
-    pm.addPass(fir::createLLVMDialectToLLVMPass(OutputFilename + ".ll"));
+    std::error_code ec;
+    llvm::ToolOutputFile out(OutputFilename + ".ll", ec,
+                             llvm::sys::fs::OF_None);
```
Comment (jeanPerier) [inline tools/bbc/bbc.cpp]: Is this change required for the tests or is this just a clean-up to align with how LLVM creates output file ?

File: test-lit/fir/fir-ops.fir (hunk 1 above)
Comment (schweitzpgi) [reply test-lit/fir/fir-ops.fir]: Yes. Yes. I believe so, but was going to check one more time.

Comment (sscalpone) [reply test-lit/fir/fir-ops.fir]: Cool.  I was asking more for a comment with a potential TODO section.

File: include/fir/Dialect/FIROps.td (hunk 2 above)
Comment (schweitzpgi) [reply include/fir/Dialect/FIROps.td]: Following the style in place like Ops.td with this.

I think there is a fine distinction drawn between a sort of generalist "reads like a tutorial" documentation (description) and practitioners "what this code is for" comments.

File: tools/bbc/bbc.cpp (hunk 3 above)
Comment (schweitzpgi) [reply tools/bbc/bbc.cpp]: It's a personal requirement since I want tools that can write to stdout to make it easier to build a pipeline for lit tests. :)

File: test-lit/fir/fir-ops.fir (hunk 1 above)
Comment (schweitzpgi) [reply test-lit/fir/fir-ops.fir]: I haven't written a (Tilikum) bridge test. That's TODO, yes.
